frontend/src/scenes/dashboard/dashboardLogic.test.ts

```diff
@@ -0,0 +1,96 @@
+import { expectLogic, initKeaTestLogic } from '~/test/kea-test-utils'
+import { dashboardLogic } from 'scenes/dashboard/dashboardLogic'
+import dashboardJson from './__mocks__/dashboard.json'
+import { truth } from '~/test/kea-test-utils/jest'
+import { dashboardLogicType } from 'scenes/dashboard/dashboardLogicType'
+import { dashboardsModel } from '~/models/dashboardsModel'
+import { dashboardItemsModel } from '~/models/dashboardItemsModel'
+import { eventUsageLogic } from 'lib/utils/eventUsageLogic'
+import { trendsLogic } from 'scenes/trends/trendsLogic'
+
+jest.mock('lib/api')
+
+describe('dashboardLogic', () => {
+    let logic: BuiltLogic<dashboardLogicType>
+
+    mockAPI(async ({ pathname, searchParams }) => {
+        if (pathname === 'api/dashboard/5/') {
+            return dashboardJson
+        } else if (pathname.startsWith('api/dashboard_item/')) {
+            return dashboardJson.items.find(({ id }) => id === parseInt(pathname.split('/')[2]))
+        } else if (pathname === '_preflight/') {
+            return { is_clickhouse_enabled: true }
+        } else {
+            throw new Error(`Unmocked fetch to: ${pathname} with params: ${JSON.stringify(searchParams)}`)
+        }
+    })
+
+    initKeaTestLogic({
+        logic: dashboardLogic,
+        props: {
+            id: 5,
+        },
+        onLogic: (l) => (logic = l),
+    })
+
+    describe('core assumptions', () => {
+        it('mounts other logics', async () => {
+            await expectLogic(logic).toMount([dashboardsModel, dashboardItemsModel, eventUsageLogic])
+        })
+
+        it('fetches dashboard items on mount', async () => {
+            expectLogic(logic)
+                .toDispatchActions(['loadDashboardItems'])
+                .toMatchValues({
+                    allItems: null,
+                    items: undefined,
+                })
+                .toDispatchActions(['loadDashboardItemsSuccess'])
+                .toMatchValues({
+                    allItems: dashboardJson,
+                    items: truth((items) => items.length === 2),
+                })
+        })
+    })
+
+    describe('reload all items', () => {
+        it('reloads when called', async () => {
+            await expectLogic(logic, () => {
+                logic.actions.refreshAllDashboardItemsManual()
+            })
+                .toDispatchActions([
+                    // starts loading
+                    'refreshAllDashboardItemsManual',
+                    'refreshAllDashboardItems',
+                ])
+                .toDispatchActionsInAnyOrder([
+                    // sets the "reloading" status
+                    logic.actionCreators.setRefreshStatus(dashboardJson.items[0].id, true),
+                    logic.actionCreators.setRefreshStatus(dashboardJson.items[1].id, true),
+                ])
+                .toMatchValues({
+                    refreshStatus: {
+                        172: { loading: true },
```

Comment:
very picky nit :) 

high coupling between the dashboard json and the test knowing that 172 and 175 are ids for items 0 and 1 

you need to read the json, lines 70 and 71, and 75 and 76 to track it

I can't see a quick way to resolve it without adding even more code to read though...

Reply:
Good point, and I agree. I do remember even adjusting this in one place after, but somehow missed this. Will adjust next time I go through this file.